go-controller/pkg/ovndbmanager/ovndbmanager.go

```diff
@@ -0,0 +1,229 @@
+package ovndbmanager
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+func RunDBChecker(stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	klog.Info("Starting DB Checker to ensure cluster membership and DB consistency")
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		ensureOvnDBState(util.OvnNbdbLocation, stopCh)
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		ensureOvnDBState(util.OvnSbdbLocation, stopCh)
+	}()
+	<-stopCh
+	klog.Info("Shutting down db checker")
+	wg.Wait()
+	klog.Info("Shut down db checker")
+}
+
+func ensureOvnDBState(db string, stopCh <-chan struct{}) {
+	ticker := time.NewTicker(60 * time.Second)
+	klog.Infof("Starting ensure routine for Raft db: %s", db)
+	_, _, err := util.RunOVSDBTool("db-is-standalone", db)
+	if err == nil {
+		klog.Info("The db is running in standalone mode")
+		return
+	} else {
+		var ee *exec.ExitError
+		if errors.As(err, &ee) {
+			klog.Infof("Exit code for the db-is-standalone: %v", ee.ExitCode())
+			if ee.ExitCode() == 1 {
```

Comment:
should we check for `!= 2` or is `== 1` check sufficient?

Reply:
i think 1 is ok, 2 means its clustered and we continue to the to the for/select loop

Reply:
I meant if we check for `!=2` that's a more tighter check. If in future the command were to include other possible exit codes, the `!=2` check will still work.

Reply:
sure that's fine too